library/ctr_drbg.c

```diff
@@ -261,10 +273,13 @@ static int ctr_drbg_update_internal( mbedtls_ctr_drbg_context *ctx,
     /*
      * Update key and counter
      */
-    mbedtls_aes_setkey_enc( &ctx->aes_ctx, tmp, MBEDTLS_CTR_DRBG_KEYBITS );
+    ret = mbedtls_aes_setkey_enc( &ctx->aes_ctx, tmp, MBEDTLS_CTR_DRBG_KEYBITS );
+    if ( ret != 0 )
+        goto exit;
     memcpy( ctx->counter, tmp + MBEDTLS_CTR_DRBG_KEYSIZE, MBEDTLS_CTR_DRBG_BLOCKSIZE );
 
-    return( 0 );
+exit:
+    return( ret );
 }
 
 void mbedtls_ctr_drbg_update( mbedtls_ctr_drbg_context *ctx,
```

Comment:
I know that its slightly pointless to check the return codes of block_cipher_df() and ctr_drbg_update_internal() as mbedtls_ctr_drbg_update() does not have a return value. However, it would be nice if we could at least abort the update if either of the calls fails. Also, it would be good to add a comment nearby explaning the situation (and perhaps raise a ticket as the API might need to be modified).

Reply:
It would be good to check the return values of of the functions ctr_drbg_update_internal() and ctr_drbg_update_internal() called from `mbedtls_ctr_drbg_reseed()` as well.

Reply:
@RonEld: I dont think this was fully addressed. There are still a couple of unchecked return values [here](https://github.com/ARMmbed/mbedtls/pull/1096/commits/d66b1b51207e05575ab939901e6c12fe8c827f4e#diff-af7ff720e446d17cba06099cb0742ec4R347) and [here](https://github.com/ARMmbed/mbedtls/pull/1096/commits/d66b1b51207e05575ab939901e6c12fe8c827f4e#diff-af7ff720e446d17cba06099cb0742ec4R352). Thanks!

Reply:
sorry, I missed this